Tasks/FileTransform/filetransform.ts

```diff
@@ -0,0 +1,26 @@
+import tl = require('vsts-task-lib/task');
+import { Package } from 'webdeployment-common/packageUtility';
+var deployUtility = require('webdeployment-common/utility.js');
+var fileTransformationsUtility = require('webdeployment-common/fileTransformationsUtility.js');
+
+async function main() {
+    var webPackage = new Package(tl.getPathInput('folderPath', true));
+    var packagePath = webPackage.getPath();
+    var JSONFiles = tl.getDelimitedInput('JSONFiles', '\n', false);
+    var XmlVariableSubstitution = tl.getBoolInput("enableXmlVariableSubstitution", false);
+    var applyFileTransformFlag = JSONFiles.length != 0 || XmlVariableSubstitution;
+    if (applyFileTransformFlag) {
+        var isFolderBasedDeployment: boolean = tl.stats(packagePath).isDirectory();
+        var folderPath = await deployUtility.generateTemporaryFolderForDeployment(isFolderBasedDeployment, packagePath, webPackage.getPackageType());
+
+        var isMSBuildPackage = !isFolderBasedDeployment && (await deployUtility.isMSDeployPackage(packagePath));
```

Comment:
Why do we have "!isFolderBasedDeployment" check. Isn't this task supposed to run with Folders only ?

Reply:
we will support zip too